File: BbApp/Model/Competition.swift
//
//  Competition.swift
//  BbApp
//
//  Created on 14/01/2026.
//

import Foundation

/// Competition model based on the OpenAPI schema.
/// Represents a sports competition with its type, gender, and category information.
///
/// Schema reference: `#/components/schemas/Competition` in `openapi.json`
struct Competition: Identifiable, Codable {
    let id: String
    let name: String
    let shortName: String?
    let sport: Sport
    let countryCode: String
    let gender: CompetitionGender?
    let category: String?
    let type: CompetitionType?
    let year: UInt16?
}

File: src/types/competition.ts
import type { Sport } from "./sport"

export type CompetitionGender = "male" | "female" | "mixed"
export type CompetitionType = "league" | "cup" | "friendly"

export interface Competition {
  id: string
  name: string
  shortName?: string
  sport: Sport
  countryCode: string
  gender?: CompetitionGender
  category?: string
  type?: CompetitionType
  year?: number
}

export interface CompetitionPhase {
  name: string
  type?: string
}

export interface CompetitionGroup {
  id: string
  name: string
  competition: Competition
  phase?: CompetitionPhase
}
